specs/agents/tracing-api-otel.md

```diff
@@ -0,0 +1,251 @@
+- Unset (default) => Unknown
+
+However, when not provided explicitly agents can infer the outcome from the presence of a reported error.
+This behavior is not expected with OTel API with status, thus bridged spans/transactions should NOT have their outcome
+altered by reporting (or lack of reporting) of an error. Here the behavior should be identical to when the end-user provides
+the outcome explicitly and thus have higher priority over the inferred value.
+
+### Attributes mapping
+
+OTel relies on key-value pairs for span attributes.
+Keys and values are protocol-specific and are defined in [semantic convention](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions) specification.
+
+In order to minimize the mapping complexity in agents, most of the mapping between OTel attributes and agent protocol will be delegated to APM server:
+- All OTel span attributes should be captured as-is and written to agent protocol.
+- APM server will handle the mapping between OTel attributes and their native transaction/spans equivalents
+- Some native span/transaction attributes will still require mapping within agents for [compatibility with existing features](#compatibility-mapping)
+
+OpenTelemetry attributes should be stored in `otel.attributes` as a flat key-value pair mapping added to `span` and `transaction` objects:
+```json
+{
+  // [...] other span/transaction attributes
+  "otel": {
+    "span_kind": "CLIENT",
+    "attributes": {
+      "db.system": "mysql",
+      "db.statement": "SELECT * from table_1"
+    }
+  }
+}
+```
+
+APM server supports the `otel.attributes` property starting with version 7.16.0.
+
+When the APM server version does not support `otel.attributes` property, agents should use `labels` as fallback with OTel attribute
+name as key.
+
+When the APM server supports `otel.attributes` property, the server-side mapping should be identical to the one
+that is applied to handle native OpenTelemetry Protocol (OTLP) intake.
+
+### Compatibility mapping
+
+Agents should ensure compatibility with the following features:
+- breakdown metrics
+- [dropped spans statistics](handling-huge-traces/tracing-spans-dropped-stats.md)
+- [compressed spans](handling-huge-traces/tracing-spans-compress.md)
+
+As a consequence, agents must provide values for the following attributes:
+- `transaction.name` or `span.name` : value directly provided by OTel API
+- `transaction.type` : see inference algorithm below
+- `span.type` and `span.subtype` : see inference algorithm below
+- `span.destination.service.resource` : see inference algorithm below
+
+#### Transaction type
+
+```javascript
+a = transation.otel.attributes;
+span_kind = transaction.otel_span_kind;
+isRpc = a['rpc.system'] !== undefined;
+isHttp = a['http.url'] !== undefined || a['http.scheme'] !== undefined;
+if (span_kind == 'SERVER' && (isRpc || isHttp)) {
+    type = 'request';
+} else if (span_kind == 'CONSUMER' && isMessaging) {
```

Comment:
how is `isMessaging` defined?

Reply:
Good question. @SylvainJuge, could you clarify?

I assume, it's `isMessaging = a['messaging.system']`.

However, that's a bit different in how APM Server would do as it looks for all known messaging attributes: https://github.com/elastic/apm-server/blob/1fad7fd216c60bc67264d70d25fa2f1a2b8e1452/processor/otel/traces.go#L610-L620

Reply:
Yes, it's exactly that @felixbarny, I've fixed that now, thanks for filling the gaps here.

Regarding the slight differences with how APM server implementation works, I don't think it's worth over-specifying the mapping behavior as the basic `messaging.system` property is expected to be present for all messaging OTel spans. The fact that the APM server will handle more corner cases is just a bonus.